photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/RandomSearch.scala

```diff
@@ -84,28 +79,16 @@ class RandomSearch[T](
         val candidate = next(lastCandidate, lastObservation)
 
         // Discretize values specified as discrete
-        discreteParams.foreach { index =>
-          candidate(index) = round(candidate(index))
-        }
+        val candidateDiscrete = discretizeCandidate(candidate, discreteParams)
```

Comment:
Minor: May be a better idea to call this candidateWithDiscrete, since candidateDiscrete might give a wrong impression.